src/plugins/data/public/index.ts

```diff
@@ -262,9 +264,12 @@ export {
   UI_SETTINGS,
   TypeMeta as IndexPatternTypeMeta,
   AggregationRestrictions as IndexPatternAggRestrictions,
+  IndexPatternSpec,
   fieldList,
 } from '../common';
 
+export * from '../common/index_patterns/errors';
```

Comment:
nit: Would be preferable to avoid an `export *` and have an explicit list of exports here, since it's the top-level `index.ts`